tests/workflows/test_main_snapshot.py

```
@@ -0,0 +1,185 @@
+                main()
+
+    def test_main_uses_default_manifest_path(self) -> None:
+        """main uses default manifest path when no argument given."""
+        mock_collect = MagicMock(return_value=[b"test\ttest\t1.0\n"])
+
+        with patch("sys.argv", ["dependency_binaries_manifest.py"]):
+            with patch.object(
+                _manifest_module,
+                "_collect_manifest_lines",
+                mock_collect,
+            ):
```

Ruff (0.15.9)

[warning] Use a single `with` statement with multiple contexts instead of nested `with` statements

(SIM117)